tasks/main.yml

```diff
@@ -37,6 +37,25 @@
     - item.value.state == 'present'
     - not item.key ~ '@' ~ item.value.domain in register_sasldblistusers.stdout_lines
 
+- name: Set cyrus_sasl_sasldb_file_real_path
+  set_fact:
+    # BSDs appends '.db' to the given file name, but others treat it as-is
+    cyrus_sasl_sasldb_file_real_path: "{% if ansible_os_family == 'FreeBSD' or ansible_os_family == 'OpenBSD' %}{{ cyrus_sasl_sasldb_file }}.db{% else %}{{ cyrus_sasl_sasldb_file }}{% endif %}"
```

Comment:
I need the explanation, too, that who decide to implement the BSD behavior. as the comment says, "$command $path" means the actual file name is "$path.db" on BSD implementations, which has been known so for years. see [dbm(3)](https://www.freebsd.org/cgi/man.cgi?dbm(3)).

as for your question, yes, as long as the library is BSD-derived one.